jsonstore/jsonstore.go

```diff
@@ -235,6 +247,35 @@ func (app *JSONStoreApplication) DeliverTx(tx []byte) types.ResponseDeliverTx {
 		}
 
 		break
+
+	case "upvoteValidator":
+		entity := body["entity"].(map[string]interface{})
+
+		// validate user exists
+		pubKeyBytes, errDecode := base64.StdEncoding.DecodeString(message["publicKey"].(string))
+
+		if errDecode != nil {
+			panic(errDecode)
+		}
+
+		publicKey := strings.ToUpper(byteToHex(pubKeyBytes))
+
+		var user User
+		err := db.C("users").Find(bson.M{"publicKey": publicKey}).One(&user)
+		if err != nil {
+			panic(err)
+		}
+
+		fmt.Println("user validated!")
+
+		// validate validator exists & update votes
+		validatorID := bson.ObjectIdHex(entity["validator"].(string))
```

Comment:
It is also possible to retrieve validatorID via the `publicKey`. This way you won't have to supply validatorID separately for voting from the front-end. Supplying just the `publicKey` of the person will suffice.

Reply:
nice. Didn't realise that. Make sense since NodeId is derived the same way

Reply:
i got back to this comment now and realised it doesnt work for me.. let's discuss it in the morning